cmd/peer-rest-server.go

```diff
@@ -1384,6 +1385,82 @@ func (s *peerRESTServer) DriveSpeedTestHandler(w http.ResponseWriter, r *http.Re
 	logger.LogIf(r.Context(), gob.NewEncoder(w).Encode(result))
 }
 
+// DevNull - everything goes to io.Discard
+func (s *peerRESTServer) DevNull(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("invalid request"))
+		return
+	}
+	for {
+		n, err := io.CopyN(io.Discard, r.Body, 128*1024)
+		atomic.AddUint64(&globalNetperfRX, uint64(n))
+		if err != nil {
+			break
+		}
+	}
+}
+
+// Reader to read random data.
+type netperfReader struct {
+	eof bool
+	n   uint64
+	buf []byte
+}
+
+func (m *netperfReader) Read(b []byte) (int, error) {
+	if m.eof {
+		return 0, io.EOF
+	}
+	n := copy(b, m.buf)
+	atomic.AddUint64(&m.n, uint64(n))
+	return n, nil
+}
+
+func netperf(ctx context.Context, duration time.Duration) madmin.NetperfNodeResult {
+	globalNetperfRX = 0
+
+	r := &netperfReader{}
+	r.buf = make([]byte, 128*1024)
+	rand.Read(r.buf)
+
+	var wg sync.WaitGroup
+	for index := range globalNotificationSys.peerClients {
+		if globalNotificationSys.peerClients[index] == nil {
+			continue
+		}
+		go func(index int) {
+			for i := 0; i < 16; i++ {
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					globalNotificationSys.peerClients[index].DevNull(ctx, r)
+				}()
+			}
+		}(index)
+	}
+
+	time.Sleep(duration)
+	r.eof = true
+	wg.Wait()
+	return madmin.NetperfNodeResult{Endpoint: "", TX: r.n / uint64(duration.Seconds()), RX: globalNetperfRX / uint64(duration.Seconds()), Error: ""}
+}
+
+// Netperf - perform netperf
+func (s *peerRESTServer) Netperf(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("invalid request"))
+		return
+	}
+
+	durationStr := r.Form.Get(peerRESTDuration)
+	duration, err := time.ParseDuration(durationStr)
+	if err != nil {
+		duration = time.Second * 10
+	}
+	result := netperf(r.Context(), duration)
```

Comment:
```suggestion
	result := netperf(r.Context(), duration.Round(time.Second))
```
Otherwise average can be wrong.

Reply:
I will sanitize duration on admin handlers instead since that is the entry point